core/src/main/java/io/grpc/internal/DnsNameResolver.java

```diff
@@ -167,7 +167,10 @@
       port = nameUri.getPort();
     }
     this.proxyDetector = Preconditions.checkNotNull(helper.getProxyDetector(), "proxyDetector");
-    this.resolveRunnable = new Resolve(this, stopwatch, getNetworkAddressCacheTtlNanos(isAndroid));
+    this.cacheTtlNanos = getNetworkAddressCacheTtlNanos(isAndroid);
+    this.stopwatch = Preconditions.checkNotNull(stopwatch, "stopwatch");
+    this.syncContext =
```

Comment:
This is a little uncomfortable to me.  It would be better if the ctor was dumber, and the DNRP constructed the dependencies for this class, rather than DNR constructing its own.  Same with proxyDetector.

Reply:
This new form was much clearer for me, because they are the same for every resolver. Having them in the resolver is harder to reason about because I then have to check if they change. I'd actually prefer if Listener was removed from Resolver as well.

Reply:
Done.